src/Grid/utils.ts

```diff
@@ -105,25 +82,98 @@ export function getBreakpointKey(
 }
 
 /*
-    Using the passed responsiveSettings, determines what responsive value should be returned for
-    the current breakpoint. Ex:
+    CSS used both by generateBreakpointCSS() and the Cell component. Aligns content vertically.
+*/
+export const middleCSS = css`
+    display: inline-flex;
+    flex-flow: column wrap;
+    justify-content: center;
+    justify-self: stretch;
+`;
 
-    const settings = { xs: 1, sm: 3, md: 6 }
-    const innerWidth = 800;
-    const sortedBreakpoints = [{ xs: 500, sm: 750, md: 1000 }];
-    getResponsiveValue(settings, innerWidth, sortedBreakpoints); // 3
+/*
+    Small helper object for the generateBreakpointCSS() function to render the correct CSS for each
+    grid setting.
 */
-export function getResponsiveValue(
-    responsiveSettings: object | number | undefined,
-    innerWidth: number,
-    sortedBreakpoints: BreakpointType[]
+const ops = {
+    // If width is 0, don't show the cell. If middle is true, add in the middle CSS.
+    width: (w: any, middle: boolean) =>
+        w > 0
+            ? middle
+                ? `grid-column-end: span ${w}; ${middleCSS}`
+                : `grid-column-end: span ${w}; display: block;`
+            : 'display:none;',
+    height: (h: any) => `grid-row-end: span ${h};`,
+    left: (l: any) => `grid-column-start: ${l};`,
+    top: (t: any) => `grid-row-start: ${t};`,
+};
+
+/*
+    Takes the gridSettings object and parses the data to generate sorted css breakpoints. It groups
+    css properties based on the breakpoint size so that a single breakpoint declaration for 'xs' can
+    have css for left, top, height & width if necessary.
+*/
+export function generateBreakpointCSS(
+    gridSettings: GridSettings,
+    sortedBreakpoints: BreakpointType[],
+    middle: boolean | undefined
 ) {
-    if (typeof responsiveSettings !== 'object') {
-        return responsiveSettings;
+    const responsiveCSS = {};
+    const sortedResponsiveCSS: BreakpointType[] = [];
+    const generalSettings: GridSettings = {};
+
+    // Loops through the props with responsive settings, i.e. left, top, etc
+    Object.keys(gridSettings).forEach((settingKey: string) => {
+        const currentValue = gridSettings[settingKey];
+
+        // If the value passed isn't an object then it's not responsive, so its value should be
+        // put into the generalSettings object
+        if (typeof currentValue !== 'object') {
+            generalSettings[settingKey] = currentValue;
+        } else {
+            Object.keys(currentValue).forEach((breakpointKey: string) => {
+                const responsiveValue = currentValue[breakpointKey];
+
+                if (responsiveCSS[breakpointKey]) {
+                    responsiveCSS[breakpointKey] += ops[settingKey](
```

Comment:
This may be more succinct.

```
if (!responsiveCSS[breakpointKey]) {
    responsiveCSS[breakpointKey] = '';
}
responsiveCSS[breakpointKey] += ops[settingKey](
    responsiveValue,
    middle
);
```